test_programs/compile_success_empty/comptime_expr/src/main.nr

```diff
@@ -0,0 +1,313 @@
+mod tests {
+    use std::meta::op::UnaryOp;
+    use std::meta::op::BinaryOp;
+
+    #[test]
+    fn test_expr_as_array() {
```

Comment:
These are all tests so maybe `noir_test_success` would be a better place for them?
All of them are run at comptime regardless though so it doesn't actually matter much.